pydantic/_internal/_generate_schema.py

```diff
@@ -1897,7 +1901,11 @@ def _callable_schema(self, function: Callable[..., Any]) -> core_schema.CallSche
             if p.annotation is sig.empty:
                 annotation = typing.cast(Any, Any)
             else:
-                annotation = type_hints[name]
+                # Note: This was originally get by `_typing_extra.get_function_type_hints`,
+                #       but we switch to simply `p.annotation` to support bultins (e.g. `sorted`).
+                #       May need to revisit if anything breaks.
+                annotation = ForwardRef(p.annotation) if isinstance(p.annotation, str) else p.annotation
+                annotation = _typing_extra.eval_type_backport(annotation, self._types_namespace)
```

Comment:
```suggestion
                annotation = self._resolve_forward_ref(annotation)
```

(some places in `GenerateSchema` still use of `eval_type_backport` but let's not fix this here for now).